### PR TITLE
Fix router `model_name` dependancy issues

### PR DESCRIPTION
**Pull request by JBKahn**

### CODE REVIEW

**Comment by JBKahn** [comment]
@4word I think I'm going to change the router to acceept not passing a model name. With the example and understanding that you have to be aware in the data migration that the `current` database may not have your model on it and I'll add a util function to check if a model is on a certain DB. I'll include an example but I'd like an opinion when I'm done.

expected to look like:

```python
def func():
    if model_on_db(some_model, migration.current_db):
        do_the_things()
Runpython(func)
```

**Comment by 4word** [comment]
Any word on a fix for this? We had to add 3 migrations on Friday, and now migrating is taking ~ 45 minutes while running unit tests (and when we actually run it outselves), way way up from the original ~10 minutes for our 500 DBs. It looks like the more migrations there are, the more the allow migrate is called and the inspect call is just super slow.

**Comment by JBKahn** [comment]
If you want, pull the latest code into your local copy and just comment out the line that does that check and return something else for now? I'm waiting for one more reply from the django team before I move forward.

**Comment by JBKahn** [comment]
@4word what do you think? I have no resolution on that issue I opened but this seems reasonable.

**Comment by JBKahn** [comment]
@4word what do you think? I have no resolution on that issue I opened but this seems reasonable.

File: tests/test_router.py [hunk 1]
```diff
@@ -316,13 +316,6 @@ def test_allow_relation_two_items_on_the_same_non_default_database(self):
         item_two = ShardedTestModelIDs.objects.create(stub=None)
         self.assertTrue(self.sut.allow_relation(item_one, item_two))
 
-    def test_not_allow_relation_two_items_on_the_different_non_default_database(self):
```
Comment (JBKahn) [inline tests/test_router.py]: there's a better test for this in the file

**Pull request merged**